pkg/util/overridemanager/commandargsoverride.go

```diff
@@ -0,0 +1,117 @@
+	case util.ReplicaSetKind:
+		fallthrough
+	case util.DeploymentKind:
+		fallthrough
+	case util.DaemonSetKind:
+		fallthrough
+	case util.StatefulSetKind:
+		return buildCommandArgsPatchesWithPath(target, "spec/template/spec/containers", rawObj, commandRunOverrider)
+	}
+	return nil, nil
+}
+func buildCommandArgsPatchesWithPath(target string, specContainersPath string, rawObj *unstructured.Unstructured, commandRunOverrider *policyv1alpha1.CommandArgsOverrider) ([]overrideOption, error) {
+	patches := make([]overrideOption, 0)
+	containers, ok, err := unstructured.NestedSlice(rawObj.Object, strings.Split(specContainersPath, pathSplit)...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieves path(%s) from rawObj, error: %v", specContainersPath, err)
+	}
+	if !ok || len(containers) == 0 {
+		return nil, nil
+	}
+	klog.V(4).Infof("buildCommandArgsPatchesWithPath containers info (%+v)", containers)
+	for index, container := range containers {
+		if container.(map[string]interface{})["name"] == commandRunOverrider.ContainerName {
+			commandArgsPath := fmt.Sprintf("/%s/%d/%s", specContainersPath, index, target)
+			commandArgsValue := make([]string, 0)
+			if container.(map[string]interface{})[target] == nil {
+				return nil, nil
+			}
+			for _, val := range container.(map[string]interface{})[target].([]interface{}) {
+				commandArgsValue = append(commandArgsValue, fmt.Sprintf("%s", val))
+			}
+			patch, err := acquireCommandArgsOverrideOption(commandArgsPath, commandArgsValue, commandRunOverrider)
+			if err != nil {
+				return nil, err
+			}
+			klog.V(4).Infof("[buildCommandArgsPatchesWithPath] containers patch info (%+v)", patch)
+			patches = append(patches, patch)
+		}
+	}
+	return patches, nil
+}
+
+func acquireCommandArgsOverrideOption(commandArgsPath string, commandArgsValue []string, commandOverrider *policyv1alpha1.CommandArgsOverrider) (overrideOption, error) {
+	if !strings.HasPrefix(commandArgsPath, pathSplit) {
+		return overrideOption{}, fmt.Errorf("internal error: [acquireCommandOverrideOption] commandRunPath should be start with / character")
+	}
+
+	newCommandArgs, err := overrideCommandArgs(commandArgsValue, commandOverrider)
+	if err != nil {
+		return overrideOption{}, err
+	}
+
+	klog.V(4).Infof("[acquireCommandArgsOverrideOption], newCommand info (%+v)", newCommandArgs)
+	return overrideOption{
+		Op:    string(policyv1alpha1.OverriderOpReplace),
+		Path:  commandArgsPath,
+		Value: newCommandArgs,
+	}, nil
+}
+
+func overrideCommandArgs(curCommandArgs []string, commandArgsOverrider *policyv1alpha1.CommandArgsOverrider) ([]string, error) {
+	var newCommandArgs []string
+	switch commandArgsOverrider.Operator {
+	case policyv1alpha1.OverriderOpAdd:
+		newCommandArgs = append(curCommandArgs, commandArgsOverrider.Value...)
+		klog.V(4).Infof("[overrideCommandArgs], op: %s, oldCommandArgs: %+v, newCommandArgs: %+v", policyv1alpha1.OverriderOpAdd, curCommandArgs, newCommandArgs)
```

Comment:
Print the args might be useful for debugging, but if there are sensitive args such as passwords, then we'll have a security issue.

Double-check here, do we really need this to be logged?